src/snowflake/connector/connection_diagnostic.py

```diff
@@ -0,0 +1,562 @@
+import base64
+import ipaddress
+import json
+import os
+import re
+import socket
+import ssl
+import tempfile
+import OpenSSL
+
+from datetime import datetime
+from logging import getLogger
+from pathlib import Path
+from typing import Any, Tuple
+from urllib.request import getproxies
+from .compat import IS_WINDOWS, urlparse
+from .cursor import SnowflakeCursor
+from .vendored import urllib3
+
+logger = getLogger(__name__)
+
+if IS_WINDOWS:
+    import winreg
+
+
+class ConnectionDiagnostic:
+    """Implementation of a connection test utility for Snowflake connector
+
+    Use new ConnectionTest() to get the object.
+    """
+
+    def __init__(self,
+        account: str,
+        host: str,
+        connection_diag_log_path: str,
+        connection_diag_whitelist_path: str,
+        proxy_host: str,
+        proxy_port: str,
+        proxy_user: str,
+        proxy_password: str,
+    ):
+        self.account = account
+        self.host = host
+        self.test_results: dict[str, Any] = {'INITIAL': [], 'PROXY': [], 'SNOWFLAKE_URL': [], 'STAGE': [],
+                                             'OCSP_RESPONDER': [], 'OUT_OF_BAND_TELEMETRY': [], 'IGNORE': []}
+        host_type: str = "INITIAL"
+        self.__append_message(host_type, f"Specified snowflake account: {self.account}")
+        self.__append_message(host_type, f"Host based on specified account: {self.host}")
+        if '.com.snowflakecomputing.com' in self.host:
+            self.host = host.split('.com.snow')[0] + ".com"
```

Comment:
It's a small trick to stop after the first split, since you only ever use the first part. For future reference you might find [partition](https://docs.python.org/3/library/stdtypes.html#str.partition) useful
```suggestion
            self.host = host.split('.com.snow', 1)[0] + ".com"
```

Reply:
Thanks!  Very helpful!